dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileSchema.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Storages/DeltaMerge/ColumnFile/ColumnFileSchema.h>
+
+
+namespace DB
+{
+namespace DM
+{
+
+Digest calcDigest(const Block & schema)
+{
+    SHA256_CTX ctx;
+    SHA256_Init(&ctx);
+    unsigned char digest_bytes[32];
+
+    const auto & data = schema.getColumnsWithTypeAndName();
+    for (const auto & column_with_type_and_name : data)
+    {
+        const auto & type = column_with_type_and_name.type->getName();
```

Comment:
How about using `getTypeId`? Hashing numbers should be efficient than hashing strings.

Reply:
We can't use getTypeId instead of getName here. 
Becuase for datatype of Nullable, 
for `getName`, it will return `std::string getName() const override { return "Nullable(" + nested_data_type->getName() + ")"; }`
however, for `getTypeId`, it wil return just `TypeIndex::Nullable`, which will make different datatype show the same id.

If we use getTypeId here,
the table `create table test.t(a decimal(20, 6))` and `create table test.t(a datetime(6))` will get the same digest, because the type id of column a is both TypeIndex::Nullable.